From patches Mon Sep 17 00:00:00 2001
From: peter.crosthwaite@xilinx.com
To: peter.crosthwaite@xilinx.com, qemu-devel@nongnu.org
Cc: peter.maydell@linaro.org, qemu-devel@nongnu.org
Date: Tue, 23 Apr 2013 12:08:01 +1000
Message-id: a1a99e3d3e17686fe0540a0e18ece49b62110275.1366682426.git.peter.crosthwaite@xilinx.com
Subject: [PATCH 1/1] target-arm: helper.c: RAZ REVIDR cp register

From: Peter Crosthwaite <peter.crosthwaite@xilinx.com>

Guests should be able to read REVIDR without suffering an abort. Just
RAZ the REVIDR register.

Signed-off-by: Peter Crosthwaite <peter.crosthwaite@xilinx.com>
---

 target-arm/helper.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/target-arm/helper.c b/target-arm/helper.c
index fd055e8..e655c74 100644
--- a/target-arm/helper.c
+++ b/target-arm/helper.c
@@ -1187,6 +1187,10 @@ void register_cp_regs_for_features(ARMCPU *cpu)
             { .name = "TLBTR",
               .cp = 15, .crn = 0, .crm = 0, .opc1 = 0, .opc2 = 3,
               .access = PL1_R, .type = ARM_CP_CONST, .resetvalue = 0 },
+            /* RAZ the REVIDR register */
+            { .name = "REVIDR",
+              .cp = 15, .crn = 0, .crm = 0, .opc1 = 0, .opc2 = 6,
+              .access = PL1_R, .type = ARM_CP_CONST, .resetvalue = 0 },
             /* crn = 0 op1 = 0 crm = 3..7 : currently unassigned; we RAZ. */
             { .name = "DUMMY",
               .cp = 15, .crn = 0, .crm = 3, .opc1 = 0, .opc2 = CP_ANY,
-- 
1.7.0.4


